src/main/java/io/kuzzle/sdk/security/KuzzleSecurity.java

```diff
@@ -764,7 +764,7 @@ public void updateProfile(@NonNull final String id, final JSONObject content, fi
         @Override
         public void onSuccess(JSONObject response) {
           try {
-            listener.onSuccess(response.getJSONObject("result").getString("_id"));
+            listener.onSuccess(new KuzzleProfile(KuzzleSecurity.this.kuzzle, response.getJSONObject("result").getString("_id"), response.getJSONObject("result")));
```

Comment:
You need to pass `result._source` instead of just `result` as a profile content.